### PR TITLE
Submission of assignment #1

### PR DESCRIPTION
**Pull request by skhokhar7**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
Updated assignment_1.ipynb notebook for Assignment #1.

## What did you learn from the changes you have made?
I learn about methods of data standardization, splitting techniques, to automate the process of best parameter value selection using GridSearchCV, data visualization, and model accuracy evaluation.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
I was thinking of manual alternate approaches, but the approaches I learned in this modules were by far superior.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
I did experience few challengers, like 'class' column removal from training and test data, and understanding the internal logic of some functions. But, with the guidance of Julia Gallucci, Edward Chen, Gayathri Girish, and Google searches I was able to figure-out solutions.

## How were these changes tested?
I executed all the notebook cell on by one and validated the results.
 
## A reference to a related issue in your repository (if applicable)
Google Searches

## Checklist
- [X] I can confirm that my changes are working as intended

### CODE REVIEW

**Comment by efantinatti** [comment]
Hi Shahid

# PR Feedback Report

**Student**: Shahid  
**GitHub Username**: @skhokhar7   
**PR Number**: 1  
**Branch**: assignment-1  
**Status**: COMPLETE  
**PR URL**: https://www.github.com/skhokhar7/lcr/pull/1

## What You Did Well (Hits)

The data was correctly inspected, with the correct number of observations, variables, response variable type and levels. The data was also correctly scaled using StandardScaler. The KNN model was initialized correctly and the parameter grid for the grid search was set up correctly. The best hyperparameter was correctly identified as 8.

## Areas for Improvement (Errors)

The grid search was performed with 5 fold cross validation instead of the required 10 fold. The model evaluation was performed on the training data instead of the test data.

## Advice and Recommendations

Your submission is well structured and most of the answers are correct. To further improve, ensure that you are using the correct cross validation folds as specified in the assignment. Additionally, double check that the model evaluation is performed on the correct data split to ensure the accuracy reflects the model's performance on unseen data.

## Overall Assessment

Good submission with a few minor errors.

**Comment by skhokhar7** [comment]
Hi E. Fantinatti,

Thanks for your feedback. I've checked my submitted assignment notebook,
following are the contents of the cell where GridSearchCV is defined. I can
see "cv=10", not sure what you are referring to. Can you please elaborate?

# Your code here...# class_train# Instantiate KNN Modelknn =
KNeighborsClassifier()
# Identify X and y axis variablesX =
class_train.drop(columns=['class'])y = class_train['class']
# Define parameter_grid for cross-validationparameter_grid = {
    'n_neighbors': range(1, 50),}
# Define GridSearchCV for cross-validationclass_tune_grid = GridSearchCV(
    estimator=knn,
    param_grid=parameter_grid,
    cv=10)


Thanks & Regards,
Shahid

**Comment by skhokhar7** [comment]
Hi E. Fantinatti,

Besides the above, I've  evaluated the model also on test data, please
advise.

# Evaluate accuracy by calling accuracy_score() function# using actual
class data from test data and predictionsaccuracy =
accuracy_score(class_test['class'],
class_predict['class'])print("Accuracy:", accuracy)


Thanks & Regards,
Shahid
